ethcontract-common/src/artifact/hardhat.rs

```diff
@@ -668,4 +848,109 @@ mod test {
         assert_eq!(a.networks.len(), 1);
         assert_eq!(a.networks["4"].address, address(0xBA));
     }
+
+    #[test]
+    fn load_multi_allow_contract_name() {
+        let artifact = HardHatLoader::new()
+            .allow_contract("A")
+            .load_from_str(Format::MultiExport, MULTI_EXPORT)
+            .unwrap();
+
+        assert_eq!(artifact.len(), 1);
+
+        let a = artifact.get("A").unwrap();
+        assert_eq!(a.name, "A");
+        assert_eq!(a.networks.len(), 2);
+        assert_eq!(a.networks["1"].address, address(0xA));
+        assert_eq!(a.networks["4"].address, address(0xAA));
+
+        let artifact = HardHatLoader::new()
+            .allow_contract("X")
+            .load_from_str(Format::MultiExport, MULTI_EXPORT)
+            .unwrap();
+
+        assert_eq!(artifact.len(), 0);
+    }
+
+    #[test]
+    fn load_multi_deny_contract_name() {
+        let artifact = HardHatLoader::new()
+            .deny_contract("A")
+            .load_from_str(Format::MultiExport, MULTI_EXPORT)
+            .unwrap();
+
+        assert_eq!(artifact.len(), 1);
+
+        let a = artifact.get("B").unwrap();
```

Comment:
Nit: given that we are explicitly denying contract "A" here. I would find this a lot less misleading if this other contract were called b. 

Looking at this more and along with my previous question. I don't even know why there are any artifacts at this point and certainly not something called "B" for any reason. Did you load the alphabet from string?

Reply:
The loaded mulit-export is the JSON constant `MULTI_EXPORT`:

https://github.com/gnosis/ethcontract-rs/blob/e5722791cc6432884981ca672c0a1f2ef20bc7d8/ethcontract-common/src/artifact/hardhat.rs#L552-L580

Reply:
Yeah, this was added in a previous PR. I didn't realise you can add a group of people as a reviewer, so I was just creating PRs without reviewers. You probably didn't get a notification, sorry about this. If you're interested, take a look at other PRs linked to #512, that will give more context about what's going on.